transformer_lens/components/abstract_attention.py

```diff
@@ -451,7 +451,9 @@ def apply_causal_mask(
             )
 
         # Index back to front to ensure local attention works
-        final_mask = self.mask[None, None, -query_ctx_length:, -key_ctx_length:]  # [1, 1, pos, pos]
+        final_mask = self.mask[
+            None, None, -query_ctx_length:, -key_ctx_length:
+        ].bool()  # [1, 1, pos, pos]
```

Comment:
The `dtype` of the mask loaded from the state dict is float, but this class expects it to be a bool tensor. Casting to bool just before applying the mask fixes all the tests.
![image](https://github.com/user-attachments/assets/23dd87b2-58ac-48dd-8b46-0854d46a7798)